server/backend/courses/models.py

```
@@ -36,18 +36,34 @@ class CourseContents(models.Model):
+
+    def clean(self):
+        if self.content_type == 'video' and not self.url:
+            raise ValidationError("Video content must have a URL.")
+        if self.content_type == 'article' and not self.text_content:
+            raise ValidationError("Article text contents cannot be empty.")
+        if self.content_type == 'pdf' and not self.url:
+            raise ValidationError("PDF content must include a url.")
+
+    def save(self, *args, **kwargs):
+        self.clean()
+        super().save(*args, **kwargs)
```

Ruff (0.8.2)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: `save` method should come before custom method

(DJ012)